### DISCUSSION TITLE
How to mock requests from the middleware using msw?

### DISCUSSION THREAD

**Discussion by illiakovalenko** [vercel/next.js#38553] [Help]
Our team used [with-msw](https://github.com/vercel/next.js/tree/canary/examples/with-msw) example in order to mock requests.
We didn't find an example of how mock requests in the middleware.

* In order to mock requests in the middleware we have to run this [code](https://github.com/vercel/next.js/blob/canary/examples/with-msw/pages/_app.js#L1-L3) from the `middleware.ts` file, as `middleware` has a different process and it's not enough to run it from the `_app.tsx`
* [msw](https://github.com/mswjs/msw) can't be used in the middleware as it uses node.js native modules, and `edge runtime` doesn't allow to use it

The question is, how to use `msw` with the `middleware`, or can someone provide a library to mock requests in the middleware?

**Comment by kristojorg** [comment]
I have submitted [a ticket in MSW](https://github.com/mswjs/msw/issues/1428) to support the edge runtime, but it looks like they won't be implementing it anytime soon. I'm hoping there is a simple way to use only the `FetchInterceptor` and thus avoid the `http` module entirely.
